src/vivarium/component.py

```diff
@@ -781,11 +781,22 @@ def _register_simulant_initializer(self, builder: "Builder") -> None:
         builder
             The builder with which to register the initializer.
         """
+        if isinstance(self.initialization_requirements, list):
+            initialization_requirements = {
+                "required_resources": self.initialization_requirements
+            }
+        else:
+            initialization_requirements = self.initialization_requirements
+            self.logger.warning(
```

Comment:
nit: use proper DeprecationWarning?

```
import warnings

warnings.warn(
    "The dict format for initialization_requirements is deprecated. "
    "You should use provide a list of the required resources.",
    DeprecationWarning,
    stacklevel=2,
)
```

Reply:
I was going to do that, and then I was reading that you need to wire this up properly to work with loguru, and I didn't want to figure out how to do that. Technically, we probably shouldn't be logging vivarium in this manner since vivarium is a library: https://loguru.readthedocs.io/en/stable/overview.html#suitable-for-scripts-and-libraries